### DISCUSSION TITLE
Limit the number of connections to a single port

### DISCUSSION THREAD

**Discussion by hack3rcon** [shadowsocks/shadowsocks-rust#1440] [Q&A]
Hello,
Is it possible to limit the number of connections to a port? For example, one port accepts 10 connections and the other port accepts 20 connections.

Thank you.

**Comment by zonyitoo** [comment]
Well, you can add a limiter right after the `accept()` call in the `server`.

**Comment by hack3rcon** [comment]
> Well, you can add a limiter right after the `accept()` call in the `server`.

Hello,
Can you show me an example?

**Reply by zonyitoo** [reply]
https://github.com/shadowsocks/shadowsocks-rust/blob/1abf4579070a71b8c5846a5a93aa0a45abfa46e4/crates/shadowsocks-service/src/server/tcprelay.rs#L68-L102

For example, add your limiter in this loop to limit TCP connections.

**Reply by hack3rcon** [reply]
Excuse me, do I have to change the source code?
My config is something like the below:
```
{
    "server": "1.2.3.4",
    "server_port": 80,
    "local_port": 1080,
    "password": "password",
    "method": "aes-256-gcm"
}
```
I just want two IP addresses to connect to port 80.

**Reply by zonyitoo** [reply]
Yes, there is no limiter builtin in this Project now.
